crates/propolis-api-types/src/lib.rs

```diff
@@ -77,15 +77,32 @@ pub struct InstanceMigrateStartRequest {
     pub migration_id: Uuid,
 }
 
-#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
-pub struct InstanceMigrateStatusRequest {
-    pub migration_id: Uuid,
+/// The status of an individual live migration.
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq, Eq)]
+pub struct InstanceMigrationStatus {
+    /// The ID of this migration, supplied either by the external migration
+    /// requester (for targets) or the other side of the migration (for
+    /// sources).
+    pub id: Uuid,
+    /// The current phase the migration is in.
+    pub state: MigrationState,
 }
 
+/// The statuses of the most recent live migrations a Propolis VM has
+/// participated in. Note that Propolis will continue reporting the status of a
+/// completed migration unless it is supplanted by a new migration in the same
+/// direction. This means it's possible for both the inbound and outbound
+/// migration statuses to be set, e.g. if a Propolis was initialized via
+/// migration in and is now the source of a migration out.
```

Comment:
It might be nice if this comment discussed the motivation behind reporting both migrations here, rather than e.g. just an `enum MigrationRole` or similar?